components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -469,7 +469,7 @@ public User getUser(String userId, Map<String, Boolean> requiredAttributes) thro
     }
 
     @Override
-    public void deleteUser(String userId) throws NotFoundException, CharonException {
+    public void deleteUser(String userId) throws NotFoundException, CharonException, BadRequestException {
```

Comment:
instead of changing the method signature can't we pass a CharonException and handle it in charon

Reply:
Even though BadRequestException is added to the method signature it is available in the interface[1]. A similar approach has been followed in [2].

[1] - https://github.com/wso2/charon/blob/master/modules/charon-core/src/main/java/org/wso2/charon3/core/extensions/UserManager.java#L67
[2] - https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/295#discussion_r507711739